### DISCUSSION TITLE
v2.2.0b0

### DISCUSSION THREAD

**Discussion by DTTerastar** [ESPresense/ESPresense#491] [Announcements]
## What's Changed
* Initial esp32c3 support, also upgrade esp32 framework to 3.5. by @DTTerastar in https://github.com/ESPresense/ESPresense/pull/489
* Radar no-motion state debounce period by @dulfer in https://github.com/ESPresense/ESPresense/pull/486

## New Contributors
* @dulfer made their first contribution in https://github.com/ESPresense/ESPresense/pull/486

**Full Changelog**: https://github.com/ESPresense/ESPresense/compare/v2.1.2...v2.2.0b0

<hr /><em>This discussion was created from the release <a href='https://github.com/ESPresense/ESPresense/releases/tag/v2.2.0b0'>v2.2.0b0</a>.</em>

**Comment by PhillipRt** [comment]
I tried to flash a Wemos C3 Mini, but the log is completely empty, and no hotspot created. ESPHome had an issue regarding wifi power with the C3 Mini, this could be related: https://github.com/esphome/esphome/pull/3405

**Reply by DTTerastar** [reply]
There is a hardware fix: https://github.com/arendst/Tasmota/discussions/15443

**Reply by chrishoage** [reply]
I have this mcu and was able to get ESPHome to work by setting the output power to 8.5db [as their documentation states](https://www.wemos.cc/en/latest/c3/c3_mini.html#about-wifi)

Is there not a way to set the output power for ESPresense?

**Reply by DTTerastar** [reply]
The issue is more how to do it.  A checkbox is kinda lame, having another firmware flavor is also lame because we're hitting the limits of the number of permutations we can build.  Not everyone w/ a c3 will want their wifi artificially lowered because of one board...

**Reply by chrishoage** [reply]
> having another firmware flavor is also lame because we're hitting the limits of the number of permutations we can build.

I fully understand this suggestion isn't an immediate solution, but perhaps something to consider for the **far** future of this project?

I wonder if the onus of firmware flavors needs to be pushed on to end users ala ESPHome? Having a "control server" that allows a user to configure the settings needing for their particular MCU, build that firmware, and flash it. Users could share their configs for their MCUs they get working in github discussions, or similar. 

It seems unfair for you as a maintainer to support a firmware for every single MCU under the sun, that's obviously impossible to maintain. 

However, it also seems that if the MCU is supported by Platform.io that it should stand to reason running ESPresense on it should be possible though setting up some manual config - like setting tx power. 

Obviously supporting RISC-V took a little but of doing, but in theory the only thing needed to allow this MCU to work is adjusting the tx power, which if there was a way to easily build firmware for end users which required these slight tweaks would be possible. 

To me it would be reasonable that any MCU that didn't have a shipped config was "unsupported" but if you got it working good for you! 

Anyway, I don't have a good answer for you. I think this project is cool, but I have these MCUs and can't use them with ESPresense but can with ESPHome  :sweat_smile: 

Thanks for your time and your work on this project :slightly_smiling_face:

**Reply by DTTerastar** [reply]
My M5 Stamp C3s Wifi is only working 1 in 3 boots as well.  Not sure why yet.
